packages/@webex/plugin-meetings/src/rtcMetrics/index.ts

```diff
@@ -60,7 +64,18 @@ export default class RtcMetrics {
       if (data.name === 'stats-report') {
         data.payload = data.payload.map(this.anonymizeIp);
       }
+
       this.metricsQueue.push(data);
+
+      // If a connection fails, send the rest of the metrics in queue and get a new connection id.
+      if (
+        data.name === 'onconnectionstatechange' &&
+        data.payload[0] &&
+        JSON.parse(data.payload[0]).value === 'failed'
+      ) {
+        this.sendMetricsInQueue();
+        this.connectionId = uuid.v4();
```

Comment:
maybe we should have a connectionId generator function split out which just calls this in case we change the scheme of generating the connectionId

Reply:
Updated